src/services/fileDestinationUrl.service.ts

```diff
@@ -0,0 +1,49 @@
+import { S3 } from 'aws-sdk';
+import { v4 as uuidv4 } from 'uuid';
+
+const createFileDestinationUrl = async ({
+  bucket,
+  fileType,
+  maxSize = 0,
+  fileName = '',
+  path = '',
+}: {
+  bucket: string;
+  fileType: string;
+  maxSize?: number;
+  fileName?: string;
+  path?: string;
+}): Promise<{
+  destinationUrl: string;
+  presignedPost: S3.PresignedPost;
+}> => {
+  const s3 = new S3();
+  const key = `${path}/${fileName || uuidv4()}`;
+  const s3Params = {
+    Bucket: bucket,
+    Fields: {
+      Key: key,
+    },
+    Expires: 3600,
+    Conditions: [
+      ['eq', '$Content-Type', fileType],
+      ['content-length-range', 0, maxSize],
+    ],
+  };
+
+  const presignedPost: S3.PresignedPost = await new Promise<S3.PresignedPost>((resolve, reject) => {
+    s3.createPresignedPost(s3Params, (err: Error|null, data) => {
```

Comment:
I found [a bug with the aws sdk](https://github.com/aws/aws-sdk-js/issues/3486) which can only be avoided if I change the way that I call `createPresignedPost` to not use a callback.

However... the sdk notes that:

```
   * @note You must ensure that you have static or previously resolved
   *   credentials if you call this method synchronously (with no callback),
   *   otherwise it may not properly sign the request. If you cannot guarantee
   *   this (you are using an asynchronous credential provider, i.e., EC2
   *   IAM roles), you should always call this method with an asynchronous
   *   callback.
```

Ultimately we have a choice:

1. Use async, which allows us to leverage EC2 IAM roles, but will cause an un-catchable fatal error if the service is deployed in an environment without apppropriate credentials.

2. Use sync, which will limit how we can set AWS credentials for this service.